common/util/FrameLimiter.cpp

```diff
@@ -16,34 +18,18 @@ FrameLimiter::FrameLimiter() {}
 
 FrameLimiter::~FrameLimiter() {}
 
-void FrameLimiter::run(double target_fps,
-                       bool experimental_accurate_lag,
-                       bool do_sleeps,
-                       double engine_time) {
-  double target_seconds;
-  if (experimental_accurate_lag) {
-    target_seconds = round_to_nearest_60fps(engine_time);
-  } else {
-    target_seconds = 1.f / target_fps;
-  }
-  double remaining_time = target_seconds - m_timer.getSeconds();
-
-  if (do_sleeps && remaining_time > 0.001) {
-    std::this_thread::sleep_for(std::chrono::microseconds(int((remaining_time - 0.001) * 1e6)));
-  }
-
-  while (remaining_time > 0) {
-    remaining_time = target_seconds - m_timer.getSeconds();
-  }
-
-  m_timer.start();
+void sleep_us(u64 us) {
+  std::this_thread::sleep_for(std::chrono::microseconds(us));
 }
 
 #else
-
 #define NOMINMAX
 #include <Windows.h>
 
+void sleep_us(u64 us) {
```

Comment:
Why have two different implementations of `sleep_us`?

Reply:
One is for linux and the other is for Windows.

Reply:
Uh, yeah. I meant why not use
```cpp
void sleep_us(u64 us) {
  std::this_thread::sleep_for(std::chrono::microseconds(us))
}
```
for both platforms?

Reply:
We need sleeps that are accurate to a few ms for this to work, but the default windows timer resolution is (sometimes) ~15 ms.  that's almost a whole frame long, so it becomes impossible to sleep for the right amount of time.  The solution is to use the windows API to request a higher timer resolution and do the sleeps manually.

The C++ standard library sleeps are kinda weird too, they have very poorly specified behavior:
"The standard recommends that a steady clock is used to measure the duration. If an implementation uses a system clock instead, the wait time may also be sensitive to clock adjustments."

and we definitely wouldn't want the frame limiter to be affected by somebody changing their system time...

Reply:
Makes sense. I did not realise that the std sleeps were so poorly defined. I always assumed they were a steady clock, because as you said, if they're not that's hugely problematic.